curp/src/client.rs

```diff
@@ -801,6 +844,43 @@ where
     }
 }
 
+/// Struct to generate
+struct BackOff {
+    /// Current timeout
+    timeout: Duration,
+    /// Max timeout
+    max_timeout: Duration,
+    /// Whether to use backoff
+    use_backoff: bool,
+}
+
+impl BackOff {
+    /// Creates a new `BackOff`
+    fn new(initial_timeout: Duration, max_timeout: Duration, use_backoff: bool) -> Self {
+        Self {
+            timeout: initial_timeout.min(max_timeout),
```

Comment:
Should we panic here? or do we allow `max timeout` < `initial timeout`?

Reply:
I added a validation in the building stage of `ClientConfig`

Reply:
Then `timeout: initial_timeout.min(max_timeout)` could be `initial_timeout` directly?

Reply:
done